critiquebrainz/db/oauth_client.py

```diff
@@ -0,0 +1,137 @@
+from critiquebrainz import db
+from critiquebrainz.db import exceptions as db_exceptions
+from critiquebrainz.utils import generate_string
+import sqlalchemy
+
+
+def create(*, user_id, name, desc, website, redirect_uri):
+    """Create new OAuth client and generates secret key for it.
+
+    Args:
+        user_id (uuid): ID of the user who manages the client.
+        name (str): Name of the client.
+        desc (str): Client description.
+        website (str): Client web site.
+        redirect_uri: URI where responses will be sent.
+    Returns:
+        Dictionary of the following structure
+        {
+            "client_id": str,
+            "client_secret": str,
+            "user_id": uuid,
+            "name": str,
+            "desc": desc,
+            "website": website,
+            "redirect_uri": redirect_uri,
+        }
+    """
+    client_id = generate_string(20)
+    client_secret = generate_string(40)
+    client = {
+        "client_id": client_id,
+        "client_secret": client_secret,
+        "redirect_uri": redirect_uri,
+        "user_id": user_id,
+        "name": name,
+        "website": website,
+        "desc": desc,
+    }
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO oauth_client VALUES(:client_id, :client_secret, :redirect_uri,
+                :user_id, :name, :desc, :website)
+        """), client)
+
+    return client
+
+
+def update(*, client_id, name=None, desc=None, website=None, redirect_uri=None):
+    """Update information related to a OAuth client
+
+    Args:
+        client_id(str): Client ID.
+        name(str, optional): Updated name of the client.
+        desc(str, optional): Updated description of the client.
+        website(str, optional): Client web site.
+        redirect_uri(str, optional): URI where responses will be sent.
+    """
+    updates = []
+    update_data = {}
+    if name is not None:
+        updates.append("name = :name")
+        update_data["name"] = name
+    if desc is not None:
+        updates.append(""" "desc" = :desc""")
```

Comment:
`"""` should be used for multi-line strings and docstrings. Here you can use `'` instead.